### PR TITLE
add simplified Chinese readme

### PR DESCRIPTION
**Pull request by growdu**

### CODE REVIEW

**Comment by MathieuBordere** [comment]
@growdu Thank you for this! @stgraber @ru-fu Do you know anyone who can review this?

**Comment by ru-fu** [comment]
@adglkh could you do a quick check?

File: README_CH.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
+# c-raft
+
+[![CI Tests](https://github.com/canonical/raft/actions/workflows/build-and-test.yml/badge.svg)](https://github.com/canonical/raft/actions/workflows/build-and-test.yml) [![codecov](https://codecov.io/gh/canonical/raft/branch/master/graph/badge.svg)](https://codecov.io/gh/canonical/raft) [![Documentation Status](https://readthedocs.org/projects/raft/badge/?version=latest)](https://raft.readthedocs.io/en/latest/?badge=latest)
+
+raft协议的完全异步C语言实现。
+
+raft库采用模块化设计，核心部分实现了完全独立于平台的raft协议逻辑。最重要的是为网络传输（发送/接收,rpc消息）和存储持久化（日志和快照）提供了可插拔的I/O接口实现。
```
Comment (adglkh) [inline README_CH.md:7]: ，核心部分实现了完全独立于平台的raft协议逻辑。重要的是为网络传输（发送/接收,rpc消息）和存储持久化（日志和快照）提供了可插拔的I/O接口实现。
->
，核心部分实现了完全独立于平台的raft算法逻辑。重要的是为网络传输（发送/接收rpc消息）和存储持久化（日志和快照）提供了可插拔的I/O接口实现。

File: README_CH.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+在使用默认选项构建库时，基于 [libuv](http://libuv.org)提供了 I/O 接口的库存实现，适合绝大多数应用场景。唯一的问题是它目前
+需要 Linux，因为它使用 Linux [AIO](http://man7.org/linux/man-pages/man2/io_submit.2.html) API 用于磁盘输入/输出。 欢迎补丁添加对更多平台的支持。
```
Comment (adglkh) [inline README_CH.md:10]: 欢迎补丁添加对更多平台的支持。
->
欢迎添加补丁以支持更多的平台。

File: README_CH.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+可以通过[raft.h](https://github.com/canonical/raft/blob/master/include/raft.h)来查看全部接口。
+
+## license
+
+这个 raft C 库是在略微修改的 LGPLv3 版本下发布的，其中包括一个版权例外，允许用户在他们的项目中静态链接库代码并按照自己的条款发布最终作品。如有需要，请查看完整[license](https://github.com/canonical/raft/blob/LICENSE)文件。
```
Comment (adglkh) [inline README_CH.md:16]: 允许用户在他们的项目中静态链接库代码并按照自己的条款发布最终作品。
->
允许用户在他们的项目中静态链接这个库的代码并按照自己的条款发布最终作品。

File: README_CH.md [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+- 日志复制
+
+- 日志压缩
+
+- 成员变更
+
+它还包括一些可选的增强功能：
+
+- 乐观流水线减少日志复制延迟
```
Comment (adglkh) [inline README_CH.md:32]: 乐观流水线减少日志复制延迟
->
优化流水线以减少日志复制的延迟

File: README_CH.md [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+- 并行写入领导者的磁盘
+- 领导者失去法定人数时自动下台
+- 领导转移延期
```
Comment (adglkh) [inline README_CH.md:35]: 领导转移延期
->
领导权转移扩展

File: README_CH.md [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+- 投票前协议
```
Comment (adglkh) [inline README_CH.md:36]: 投票前协议
->
预投票协议

File: README_CH.md [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+
+理解如何使用raft库的最好方式是阅读源码目录下的[example server](https://github.com/canonical/raft/blob/master/example/server.c)。
+
+可以运行如下命令来了解example server的运行情况：
+
+```bash
+./example/cluster
+```
+
+这个命令启动了一个3节点组成的集群，并且会不断的随机停止一个节点并重启。
```
Comment (adglkh) [inline README_CH.md:75]: 这个命令启动了一个3节点组成的集群，并且会不断的随机停止一个节点并重启。
->
这个命令启动了一个由3节点组成的集群，并且会不断的随机停止一个节点并重启。

File: README_CH.md [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+raft_apply(&raft, &req, &buf, 1, apply_callback);
+```
+
+7. 添加更多的服务器节点到集群使用```raft_add()``` 和```raft_promote``` APIs
+
+## Usage Notes
+
+默认基于 [libuv](http://libuv.org) 的 `raft_io` 实现使用 `liblz4 `库压缩 raft 快照。 除了节省磁盘空间，`lz4 `压缩快照以[内容校验和](https://github.com/lz4/lz4/blob/dev/doc/lz4_Frame_format.md)的形式提供额外的数据完整性检查，这允许 `raft`检测存储期间发生的损坏。 因此，建议不要通过`--disable-lz4` 配置标志禁用 `lz4 `压缩。
+
+当启动时设置环境变量` LIBRAFT_TRACE` 时，将启用详细跟踪。
```
Comment (adglkh) [inline README_CH.md:152]: 当启动时设置环境变量` LIBRAFT_TRACE` 时，将启用详细跟踪。
->
当环境变量`LIBRAFT_TRACE`在启动时被设置，将启用详细跟踪。

**Review by adglkh — COMMENTED** [review]
Just a few nitpicks.
Overall, it looks great. Thanks!.

**Comment by MathieuBordere** [comment]
I'm just a bit worried that we probably aren't able to maintain this regularly, i.e. keeping it in line with the English `README`.

**Comment by adglkh** [comment]
LGTM, thanks!

**Comment by adglkh** [comment]
@MathieuBordere +1, that's a valid concern. To be honest, having a simplified Chinese version of the readme file is really something "nice to have". I'm not quite sure how much effort we're going to take to maintain this regularly. If anything you want from me, I'm happy to help anytime.

**Comment by freeekanayaka** [comment]
I quickly looked at the content of this new chinese README using Google Translate.

I somehow share the same concern as @MathieuBordere about being able to maintain this Chinese version of the README. One suggestion that I have is to just keep the parts that are "generic" enough that are unlikely to change over time, like the initial introductory paragraph, plus the License, Features and Credits section. A note should be also added to the Chinese version that for additional details the user should look at the English version of the README.

My 2c.

**Comment by MathieuBordere** [comment]
@growdu Canonical requires new contributors to sign the Canonical CLA [here](https://ubuntu.com/legal/contributors).
It's a bit of a hassle, but if you do it once, you can contribute to all Canonical repos afterwards.

Can you also add a note, like @freeekanayaka suggests, that the Chinese `README_CH` can be outdated in comparison with the English `README` and that users should refer to the English `README` for the most up-to-date information?

**Comment by growdu** [comment]
> @growdu Canonical requires new contributors to sign the Canonical CLA [here](https://ubuntu.com/legal/contributors). It's a bit of a hassle, but if you do it once, you can contribute to all Canonical repos afterwards.
> 
> Can you also add a note, like @freeekanayaka suggests, that the Chinese `README_CH` can be outdated in comparison with the English `README` and that users should refer to the English `README` for the most up-to-date information?

okay，I have done these.

**Pull request merged**